### PR TITLE
refactor: remove ctx.jsonp and add egg-jsonp plugin

### PR DESCRIPTION
**Pull request by dead-horse**
##### Checklist

- [x] `npm test` passes
- [x] tests and/or benchmarks are included
- [ ] documentation is changed or added
- [x] commit message follows commit guidelines

##### Description of change

Closes https://github.com/eggjs/egg/issues/288

### CODE REVIEW

File: test/fixtures/apps/secure-app/app/router.js [hunk 1]
```diff
@@ -1,4 +1,4 @@
 module.exports = app => {
-  app.get('/user.json', app.controller.index.getUser);
+  app.get('/user.json', app.jsonp(), app.controller.index.getUser);
```
Comment (atian25) [inline test/fixtures/apps/secure-app/app/router.js]: 顺手改为 `index.getUser` 吧

File: test/fixtures/apps/demo/app/router.js [hunk 2]
```diff
@@ -6,8 +6,8 @@ module.exports = app => {
     this.body = this.protocol;
   });
 
-  app.get('/user.json', function*() {
-    this.jsonp = { name: 'fengmk2' };
+  app.get('/user.json', app.jsonp(), function*() {
+    this.body = { name: 'fengmk2' };
   });
   app.get('/ip', app.controller.ip);
```
Comment (atian25) [inline test/fixtures/apps/demo/app/router.js]: 顺手改为 `'ip'` 吧

Comment (dead-horse) [inline test/fixtures/apps/demo/app/router.js]: 这个 demo 不知道是用来测啥的，先不改了，secure 那个可以改掉

Comment (popomore) [inline test/fixtures/apps/demo/app/router.js]: 这个 demo 应该是默认的，一堆测试在用

**Pull request merged**